.github/workflows/benchmark.yml

```
@@ -0,0 +1,32 @@
+name: benchmarks
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+    paths-ignore:
+      - 'LICENSE.md'
+      - 'README.md'
+      - 'docs/**'
+      - 'format/**'
+      - 'test/**'
+jobs:
+    benchmarks:
+        runs-on: ubuntu-latest
+        strategy:
+          fail-fast: false
+        steps:
+          - uses: actions/checkout@v3
+          - uses: julia-actions/setup-julia@v1
+            with:
+              version: "1"
+          - uses: julia-actions/cache@v1
+          - uses: julia-actions/julia-buildpkg@v1
+          - run: |
+              git fetch origin +:refs/remotes/origin/HEAD
+              julia --project=benchmark/ -e 'using Pkg; Pkg.instantiate(); Pkg.develop(PackageSpec(path=pwd()))'
+              # Pkg.update() allows us to benchmark even when dependencies/compat requiremetns change
+              julia --project=benchmark/ -e 'using PkgBenchmark, Loess; export_markdown(stdout, judge(Loess, BenchmarkConfig(;id="origin/HEAD", juliacmd=`julia -O3 -e"using Pkg; Pkg.update(); verbose=true))'
```

```suggestion
              julia --project=benchmark/ -e '
                  using PkgBenchmark, Loess
                  juliacmd = `$(Base.julia_cmd()) -O3 --project=$(Base.active_project()) -e "using Pkg; Pkg.update()"`
                  config = BenchmarkConfig(; id="origin/HEAD", juliacmd)
                  export_markdown(stdout, judge(Loess, config; verbose=true))'
```
There were a couple of terminators missing from the `juliacmd` definition, which have been included as part of this suggestion. Subprocesses don't inherit the parent's active project, so that will need to be set explicitly in the command.

Are you certain passing an `--eval` to the Julia command works as intended here? I don't know PkgBenchmark but if it runs code as a program file then this will not work, but if it's via subsequent `--eval`s then it should. A cursory glance at the PkgBenchmark documentation didn't reveal any similar examples or relevant info.

---

I borrowed this from a mixture of TimeZones.jl's config and one I created for MixedModels, which works:

https://github.com/JuliaStats/MixedModels.jl/pull/710

---

I think PkgBenchmark handles setting the project.

```suggestion
              julia --project=benchmark/ -e '
                  using PkgBenchmark, Loess
                  juliacmd = `$(Base.julia_cmd()) -O3 -e "using Pkg; Pkg.update()"`
                  config = BenchmarkConfig(; id="origin/HEAD", juliacmd)
                  export_markdown(stdout, judge(Loess, config; verbose=true))'
```